Documentation/project-docs/library-servicing.md

```diff
@@ -4,23 +4,34 @@ This document provides the steps necessary after modifying a CoreFx library in a
 
 ## Determine AssemblyVersion
 
-Much like `PackageVersion`, each library package has a property called `AssemblyVersion`. This will be in either the `.csproj` or in `Directory.Build.Props`/`dir.props`. If your library's `.csproj` (inside the library's `src` directory) references `.NETStandard` or `.NETFramework`, the `AssemblyVersion` should have its revision incremented by 1 (e.g. `4.0.0.0` -> `4.0.0.1`). If we don't update the AssemblyVersion, a user could run into a situation where two different dependencies of their application pull a version of your library with the same `AssemblyVersion` (`4.0.0.0`), but different `PackageVersion`. If the older package is loaded first, then the application will be using the older version of your library.
+Each library has a property called `AssemblyVersion` which will be in either the library `.csproj` or in `Directory.Build.Props` (or dir.props). For servcing events you will want to increment the revision by 1 (e.g `4.0.0.0` -> `4.0.0.1`) in the library's `Directory.Build.Props` (or dir.props) if the library ships in its own package and has an asset that is applicable to .NET Framework. To determine if it applies to .NET Framework you should check to see if there are any `netstandard` or `netfx`/`net4x` configurations in the ref and/or src project, if there are then it has assets that apply.
+
+The reason we need to increment the assembly version for things running on .NET Framework is because of the way binding works there. If there are two assemblies with the same assembly version the loader will essentially pick the first one it finds and use that version so applications don't have full control over using the later build with a particular fix included. This is worse if someone puts the older assembly in the GAC as the GAC will always win for matching assembly versions so an application couldn't load the newer one because it has the same assembly version.
+
+If this library ships both inbox on a platform and in its own library package then we need to keep the reference assembly version pinned to the same version that was shipped inbox on. In those cases we generally need to condition the AssemblyVersion in the library `ref` project like such:
+
+```
+    <!-- Must match version supported by frameworks which support 4.0.* inbox.
+         Can be removed when API is added and this assembly is versioned to 4.1.* -->
+    <AssemblyVersion Condition="'$(TargetsNetFx)' != 'true'">4.0.3.0</AssemblyVersion>
+```
+Where the `AssemblyVersion` is set to the old version before updating. To determine if the library ships inbox you can look at for `InboxOnTargetFramework` item groups or `TreatAsOutOfBox` suppressions in the pkgproj for the library.
```

Comment:
TreatAsOutOfBox isn't really relevant here.  It's actually doing the reverse, letting you continue to apply an out-of-box asset (and thus likely necessitating a version bump).

Reply:
The DiagnosticSource issue I just hit was a case where TreatAsOutOfBox was set but I needed to pin the version for the ref for everything but .NET Framework. I was trying to figure out some clear guidance to determine this but everything I came up with was a little too complicated.

Reply:
For DiagnosticSource we needed to pin because of the patch-family issue, that's a different pivot entirely and one that I consider still "open".

Reply:
It was patch family but it wouldn't have been an issue if it wasn't inbox. So it is both to help avoid moving the package dependency but it is also to prevent ref-def mismatches with things inbox. What would you suggest as a good test for folks to decide whether or not they need to pin the ref?